### PR TITLE
POM for newCalendar() (only first test)

### PR DESCRIPTION
**Pull request by SVar25**
https://trello.com/c/y3tt7UKK/509-entities-calendar-input-pom-implementation

### CODE REVIEW

File: src/test/java/EntityCalendarTest.java [hunk 1]
```diff
@@ -15,50 +18,34 @@
...
 
-        WebDriver driver = getDriver();
-
-        WebElement calendar = driver.findElement(By.xpath("//p[contains(text(),'Calendar')]"));
-        ProjectUtils.click(driver, calendar);
-
-        WebElement newCalendar = driver.findElement(By.xpath("//div[@class='card-icon']/i"));
-        newCalendar.click();
+        MainPage mainPage = new MainPage(getDriver());
+        CalendarEditPage calendarEditPage = mainPage.clickMenuCalendar().clickNewButton();
```
Comment (Stan575) [inline src/test/java/EntityCalendarTest.java:33]: can be one line

File: src/test/java/EntityCalendarTest.java [hunk 2]
```diff
@@ -15,50 +18,34 @@
...
-
-        WebElement dateElement = driver.findElement(By.xpath("//input[contains(@name, 'date')]"));
-        dateElement.click();
-        Actions actions = new Actions(driver);
-        actions.moveToElement(dateElement).build().perform();
-
-        WebElement dateTimeElement = driver.findElement(By.xpath("//input[contains(@name, 'datetime')]"));
+        WebElement dateTimeElement = getDriver().findElement(By.xpath("//input[contains(@name, 'datetime')]"));
+        dateTimeElement.click();
         dateTimeElement.click();
```
Comment (Stan575) [inline src/test/java/EntityCalendarTest.java:39]: selenium code is not allowed here, should be in a page class

File: src/test/java/EntityCalendarTest.java [hunk 3]
```diff
@@ -15,50 +18,34 @@
...
-        Actions actions1 = new Actions(driver);
-        actions1.moveToElement(dateTimeElement).build().perform();
 
-        WebElement submit = driver.findElement(By.xpath("//button[text() = 'Save']"));
-        ProjectUtils.click(driver, submit);
+        calendarEditPage.clickSaveButton();
 
         WebElement listElement =
                 getWebDriverWait().until(ExpectedConditions.elementToBeClickable(By.xpath("//div[2]/div[1]//div[1]/div/ul/li[2]/a")));
         listElement.click();
```
Comment (Stan575) [inline src/test/java/EntityCalendarTest.java:45]: same thing about selenium code

File: src/test/java/EntityCalendarTest.java [hunk 4]
```diff
@@ -15,50 +18,34 @@
...
 
-        driver.findElement(By.xpath("//div[contains(text(), '" + STRING + "')]"));
+        Assert.assertTrue(getDriver().findElement(By.xpath("//div[contains(text(), '" + STRING + "')]")).isDisplayed());
+        //Assert.assertTrue(getDriver().findElement(By.xpath("//tr[@data-index='0']//i[contains(@class, 'fa fa-check-square-o')]")).isDisplayed());
```
Comment (Stan575) [inline src/test/java/EntityCalendarTest.java:48]: we have strict rules about commented out code

**Review by Stan575 — CHANGES_REQUESTED** [review]
check the comments, the rest well done

**Comment by SVar25** [comment]
Sent from Mail for Windows 10 From: Stan AretinskiySent: Saturday, January 23, 2021 12:27 PMTo: SergeiDemyanenko/PlatformaticaQACc: Svitlana Varakuta; AuthorSubject: Re: [SergeiDemyanenko/PlatformaticaQA] POM for newCalendar() (only first test) (#521) @Stan575 requested changes on this pull request.check the comments, the rest well doneIn src/test/java/EntityCalendarTest.java:> +        MainPage mainPage = new MainPage(getDriver());+        CalendarEditPage calendarEditPage = mainPage.clickMenuCalendar().clickNewButton();can be one lineI will fix itIn src/test/java/EntityCalendarTest.java:> +        WebElement dateTimeElement = getDriver().findElement(By.xpath("//input[contains(@name, 'datetime')]"));+        dateTimeElement.click();         dateTimeElement.click();selenium code is not allowed here, should be in a page classI cant do it because the datetime element in the calendar is only populated with .click or Action.  I tried many times by analogy with data, but always the test failed. In src/test/java/EntityCalendarTest.java:>          WebElement listElement =                 getWebDriverWait().until(ExpectedConditions.elementToBeClickable(By.xpath("//div[2]/div[1]//div[1]/div/ul/li[2]/a")));         listElement.click();same thing about selenium codeI will fix it In src/test/java/EntityCalendarTest.java:>           WebElement listElement =                 getWebDriverWait().until(ExpectedConditions.elementToBeClickable(By.xpath("//div[2]/div[1]//div[1]/div/ul/li[2]/a")));         listElement.click(); -        driver.findElement(By.xpath("//div[contains(text(), '" + STRING + "')]"));+        Assert.assertTrue(getDriver().findElement(By.xpath("//div[contains(text(), '" + STRING + "')]")).isDisplayed());+        //Assert.assertTrue(getDriver().findElement(By.xpath("//tr[@data-index='0']//i[contains(@class, 'fa fa-check-square-o')]")).isDisplayed());we have strict rules about commented out codeI will fix it —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by Stan575** [comment]
@SVar25 should this PR be closed?

**Review by Seid31 — APPROVED** [review]

**Pull request closed without merging**